src/vehicle_gui.cpp

```diff
@@ -2731,6 +2732,15 @@ struct VehicleViewWindow : Window {
 		}
 	}
 
+	void OnMouseOver(Point pt, int widget) override
+	{
+		bool start_stop = widget == WID_VV_START_STOP;
+		if ((start_stop && !mouse_over_start_stop) || (!start_stop && mouse_over_start_stop)) {
+			mouse_over_start_stop = mouse_over_start_stop ? false : true;
```

Comment:
You're just testing if the mouse-is-over-the-widget state changed and setting the flag to the new state. Then marking the widget dirty only when the state changed. No need to overcomplicate it.
```suggestion
		if (start_stop != mouse_over_start_stop) {
			mouse_over_start_stop = start_top;
```

Reply:
Ah, so simple! I spent 15 minutes trying to simplify it but wasn't able to.